docs/upgrades/v0.14-to-v1-upgrade/upgraded-examples/src/test/kotlin/examples/ErrorHandlingChanges.kt

```diff
@@ -0,0 +1,237 @@
+ * high quality error messages.
+ *
+ * Also [PErrorListener] is introduced to allow customized error handling. You can register a [PErrorListener] in context of
+ * major PartiQL components [PartiQLParser], [PartiQLPlanner] and [PartiQLCompiler]. It allows you to inspect each PError
+ * the component emitted and decide action on error. You may treat all warnings as errors, suppress all warnings, delay the error handling
+ * or throw custom exception with an error message.
+ *
+ * Note for the output when you choose to delay the error handling. The component will not stop and still return an output object
+ * In case the severity is [Severity.ERROR], the output may be incorrect or incomplete. You should discard the result.
+ * You may proceed with [Severity.WARNING], but be aware of the warning condition such as date accuracy loss.
+ */
+class ErrorHandlingChanges {
+    class CustomPErrorListener : PErrorListener {
+        val errorCollection: MutableList<PError> = mutableListOf()
+
+        override fun report(error: PError) {
+            when (error.severity.code()) {
+                // You may choose to customize the error message and then halt the flow by throwing the error, or you can record the error and then throw when it reaches maximum count
+                // In this case, we record only without throwing
+                Severity.ERROR -> errorCollection.add(error)
+                // You may choose to customize the error message, treat warning as an error and then halt the flow by throwing, or you can suppress all warnings.
+                // In this case, we record only without throwing
+                Severity.WARNING -> errorCollection.add(error)
+                else -> error("This shouldn't have occurred.")
+            }
+        }
+    }
+
+    /**
+     * The example shows the error handling when using [PartiQLParser] to parse a string statement,
+     * When an error occurs, the parser will stop the parsing process by throwing the [PRuntimeException].
+     */
+    @Test
+    fun `PartiQLParser default error handling`() {
+        val parser = PartiQLParser.standard()
+
+        // The PRuntimeException is expected to throw when .parse called on a PartiQL query with invalid syntax.
+        val exception = assertThrows<PRuntimeException> {
+            try {
+                // Extra single quote is added
+                parser.parse("SELECT * FROM 'mytable;")
+            } catch (e: PRuntimeException) {
+                // Rethrow for validating the exception in test
+                throw e
+            }
+        }
+
+        assertEquals(exception.error.code(), PError.UNEXPECTED_TOKEN)
+        assertEquals(PErrorKind.SYNTAX, exception.error.kind.code())
+        assertEquals(Severity.ERROR, exception.error.severity.code())
+        assertEquals(SourceLocation(1, 15, 1), exception.error.location)
```

Comment:
Let's add some comments to highlight that in v1 (as opposed to v0.14.9), we can extract more relevant error information without needing to do string parsing.